test/EFCore.PG.FunctionalTests/Query/ArrayArrayQueryTest.cs

```diff
@@ -363,13 +361,74 @@ public void Index_text_with_constant()
 
         #endregion
 
+        #region ArrayToString
+
+        [Fact]
+        public void String_Join_Array()
+        {
+            using var ctx = CreateContext();
+
+            var found = ctx.SomeEntities.FirstOrDefault(x => string.Join(" ", x.SomeArray) == "3 4");
+            Assert.NotNull(found);
+            Assert.Contains("array_to_string", Fixture.TestSqlLoggerFactory.Sql);
+        }
+
+
+        [Fact]
+        public void Array_to_tsvector_concat()
+        {
+            using var ctx = CreateContext();
+
+            var found = ctx.SomeEntities.FirstOrDefault(x => EF.Functions.ToTsVector(string.Join(" ", x.SomeArray))
+                .Matches("3 & 4"));
+
+            var notFound = ctx.SomeEntities.FirstOrDefault(x => EF.Functions.ToTsVector(string.Join(" ", x.SomeArray))
+                .Matches("3 & 5"));
+
+            Assert.NotNull(found);
+            Assert.Null(notFound);
+            Assert.Contains("array_to_string", Fixture.TestSqlLoggerFactory.Sql);
+        }
+
+        #endregion
+
+        #region IEnumerable Join tests
+
+        [Fact]
+        public void StringArray_Join()
+        {
+            using var ctx = CreateContext();
+
+            var found = ctx.SomeEntities.FirstOrDefault(x => x.SomeStringArray.Join(" ") == "one two three");
```

Comment:
As above, these use an internal extension method, we shouldn't be translating this.